dbms/programs/client/Client.cpp

```diff
@@ -185,12 +189,120 @@ class TestHint
     }
 };
 
-
 class Client : public Poco::Util::Application
 {
 public:
     Client() {}
 
+    void init_suggestions(Completion::TSTNode &node)
+    {
+        QUERYPART *qP = queryParts;
+        while (qP->name) {
+            node.add_word(qP->name);
+            qP++;
+        }
+
+        std::vector<Block> blocks;
+
+        //preload all functions, table engines, formats, table functions, data types, combinators
+        sendQuery(
+                "SELECT name FROM system.functions"
+                " UNION ALL "
+                "SELECT name FROM system.table_engines"
```

Comment:
Will client be able to work with older versions of server, which doesn't have those tables?